inner-scripts/kubestellar

```diff
@@ -323,10 +372,45 @@ function ensure_espw() {
     else
         kubectl ws create "$espw_name" --enter
     fi
+    run_dex
+    sleep 2
+    run_kb_backend
+    kubectl apply -f "$bindir/../config/crds/"
+    kubectl get crd -oname | grep kubestellar.io | xargs -i kubectl label {} kube-bind.io/exported=true
     eval kubectl apply -f "$bindir/../config/exports" $verbdir
     echo "Finished populating the espw with kubestellar apiexports"
 }
 
+function run_dex() {
+    echo "--< starting dex in background >--"
+    dex serve "$bindir/../dex/dex-config-dev.yaml" &
+
+    run_status=$(wait_for_process dex)
+    if [ $run_status -eq 0 ]; then
+        echo " dex started"
+    else
+        echo "dex not started"
+        sleep 2
+        exit 1
+    fi
+}
+
+function run_kb_backend() {
+    echo "--< starting kube-bind example-backend in background >--"
+    kubectl ws root:espw
+    kubectl apply -f "$bindir/../kube-bind/deploy/crd"
+    example-backend --oidc-issuer-client-id=kube-bind --oidc-issuer-client-secret=ZXhhbXBsZS1hcHAtc2VjcmV0 --oidc-issuer-url=http://127.0.0.1:5556/dex --oidc-callback-url=http://127.0.0.1:8080/callback --pretty-name="BigCorp.com" --namespace-prefix="kube-bind-" --cookie-signing-key=bGMHz7SR9XcI9JdDB68VmjQErrjbrAR9JdVqjAOKHzE= --cookie-encryption-key=wadqi4u+w0bqnSrVFtM38Pz2ykYVIeeadhzT34XlC1Y= --consumer-scope="Cluster" &> /dev/null &
+
+    run_status=$(wait_for_process example-backend)
+    if [ $run_status -eq 0 ]; then
+        echo " example-backend started"
+    else
+        echo "example-backend not started"
+        sleep 2
```

Comment:
What is this sleep for?

Reply:
I remember @ezrasilvera used this line to help address some timing issue.